### PR TITLE
Added RestController, removed unused imports, made some improveents

### PR DESCRIPTION
**Pull request by Alexey007gold**

### CODE REVIEW

File: basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 1]
```diff
@@ -18,14 +18,14 @@
     @Autowired
     protected HandlerFactory<RequestHandler> handlerFactory;
 
-    private Class<T> clazz;
+    private static Class clazz;
 
     @Override
     public Response<?> handle(Request<?> msg) {
-        clazz = (Class<T>) ((ParameterizedType) this.getClass().getGenericSuperclass()).getActualTypeArguments()[0];
+        clazz = (Class) ((ParameterizedType) this.getClass().getGenericSuperclass()).getActualTypeArguments()[0];
```
Comment (Dominys) [inline basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Alexey007gold move to static

File: frontend/src/main/java/com/softgroup/RestController.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Created by alex on 24.02.17.
+ */
+
+@org.springframework.web.bind.annotation.RestController
+@RequestMapping(path = "",
+        method = RequestMethod.POST,
+        consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE)
+public class RestController {
```
Comment (Dominys) [inline frontend/src/main/java/com/softgroup/RestController.java]: @Alexey007gold make separate module resf-frontend for this. There will be one more frontend module : ws-frontend

File: frontend/src/main/java/com/softgroup/RestController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Autowired
+    private HandlerFactory<RouterHandler> handlerFactory;
+
+    @RequestMapping(path = "")
+    public Response<?> getRequest(@RequestBody final Request<?> request) {
+        try {
+            return handlerFactory.getHandler("main").handle(request);
```
Comment (Dominys) [inline frontend/src/main/java/com/softgroup/RestController.java]: @Alexey007gold It's not supposed to be here. You should have some first fouter. What is main? Nobody knows what is it.

File: frontend/src/main/java/com/softgroup/configuration/RestApplicationConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.FilterType;
+
+/**
+ * Created by alex on 16.02.17.
+ */
+
+@Configuration
+@ComponentScan(basePackages = "com.softgroup",
```
Comment (Dominys) [inline frontend/src/main/java/com/softgroup/configuration/RestApplicationConfiguration.java]: @Alexey007gold this configuration should scan only this package. Other packages must have their own configurations.

File: basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java (hunk 1 above)
Comment (Alexey007gold) [reply basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: Do you mean assigning value in the place of declaration? Like this?
private static Class clazz = (Class) ((ParameterizedType) this.getClass().getGenericSuperclass()).getActualTypeArguments()[0];

Comment (Dominys) [reply basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Alexey007gold yes. private static final. Value of this field will not change on runtime, thats first.
Next - if you will use reflection call on each request performance will be low.

Comment (Alexey007gold) [reply basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: As I see, we cant store this in one static variable, cause we need different types, as LoginRequest, or ContactSyncRequest. And static even cant be initialized this way without actual object.

Comment (Dominys) [reply basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Alexey007gold Yes. Sorry. Not static, but private final. And make field Class< T >

File: basis/common-exceptions/src/main/java/com/alex_koveckiy/common/exceptions/configuration/ModuleConfig.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Created by alex on 01.03.17.
+ */
+
+@Configuration
+@ComponentScan(basePackages = "com.alex_koveckiy.common.exceptions")
+public class ModuleConfig {
```
Comment (Dominys) [inline basis/common-exceptions/src/main/java/com/alex_koveckiy/common/exceptions/configuration/ModuleConfig.java]: @Alexey007gold Why you need configurations for this module? There is no spring components in this module.

File: basis/common-protocol/src/main/java/com/alex_koveckiy/common/protocol/configuration/ModuleConfig.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Created by alex on 01.03.17.
+ */
+
+@Configuration
+@ComponentScan(basePackages = "com.alex_koveckiy.common.protocol")
```
Comment (Dominys) [inline basis/common-protocol/src/main/java/com/alex_koveckiy/common/protocol/configuration/ModuleConfig.java]: @Alexey007gold Why you need configuration for protocol module ?

File: basis/common-router/src/main/java/com/alex_koveckiy/common/router/api/HandlerFactoryImpl.java [hunk 7]
```diff
@@ -20,14 +20,18 @@
 
     private Map<String, T> handlerMap = new HashMap<>();
 
+    public HandlerFactoryImpl() {
+        int a = 5;
```
Comment (Dominys) [inline basis/common-router/src/main/java/com/alex_koveckiy/common/router/api/HandlerFactoryImpl.java]: @Alexey007gold O_o ?

File: basis/common-router/src/main/java/com/alex_koveckiy/common/router/configuration/ModuleConfig.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+
+/**
+ * Created by alex on 01.03.17.
+ */
+
+@Configuration
+@ComponentScan(basePackages = "com.alex_koveckiy.common.router")
+public class ModuleConfig {
```
Comment (Dominys) [inline basis/common-router/src/main/java/com/alex_koveckiy/common/router/configuration/ModuleConfig.java]: @Alexey007gold Name this accordingly to module name

File: frontend/rest-frontend/pom.xml [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>rest-frontend</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <packaging>war</packaging>
+
+    <dependencies>
+        <dependency>
```
Comment (Dominys) [inline frontend/rest-frontend/pom.xml:19]: @Alexey007gold Add this to dependencyManagement of parent module and set versions there

File: frontend/rest-frontend/src/main/java/com/alex_koveckiy/frontend/rest/MyRestController.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+@RestController
+@RequestMapping(path = "",
+        method = RequestMethod.POST,
+        consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE)
+public class MyRestController {
+
+    @Autowired
+    private HandlerFactory<RouterHandler> handlerFactory;
```
Comment (Dominys) [inline frontend/rest-frontend/src/main/java/com/alex_koveckiy/frontend/rest/MyRestController.java]: @Alexey007gold Why factory? You need first router here.

File: services/authorization/authorization-api/src/main/java/com/alex_koveckiy/authorization/api/configuration/ModuleConfig.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Created by alex on 01.03.17.
+ */
+
+@Configuration
+@ComponentScan(basePackages = "com.alex_koveckiy.authorization.api")
```
Comment (Dominys) [inline services/authorization/authorization-api/src/main/java/com/alex_koveckiy/authorization/api/configuration/ModuleConfig.java]: @Alexey007gold ?

**Comment by Dominys** [comment]
@Alexey007gold underscore names in java packages is bad idea. Use appropriate name accordingly to Java code notation. Next - 78 files...

**Comment by Dominys** [comment]
@Alexey007gold Split this to different PRs.

**Pull request closed without merging**